src/scripts/main.js

```diff
@@ -1,3 +1,283 @@
+  .map(columnName => columnName.innerText);
+
+const body = document.querySelector('tbody');
+const rows = [...body.querySelectorAll('tr')];
+const getCellContent = (row, n) =>
+  row.children[n].textContent.replace(/[$,]/g, '');
+
+headList.addEventListener('click', (e) => {
+  let sorted;
+  const target = e.target.closest('th').innerText;
+  const numOfColum = headName.indexOf(target);
+
+  if (numOfColum >= 3) {
+    sorted = rows.sort((a, b) =>
+      getCellContent(a, numOfColum) - (getCellContent(b, numOfColum)));
+  } else {
+    sorted = rows.sort((a, b) =>
+      getCellContent(a, numOfColum).localeCompare(
+        getCellContent(b, numOfColum)));
+  };
+  sorted.map(i => body.appendChild(i));
+});
+
+headList.addEventListener('dblclick', (e) => {
+  let sorted;
+  const target = e.target.closest('th').innerText;
+  const numOfColum = headName.indexOf(target);
+
+  if (numOfColum >= 3) {
+    sorted = rows.sort((a, b) =>
+      getCellContent(b, numOfColum) - (getCellContent(a, numOfColum)));
+  } else {
+    sorted = rows.sort((a, b) =>
+      getCellContent(b, numOfColum).localeCompare(
+        getCellContent(a, numOfColum)));
+  };
+
+  sorted.map(i => body.appendChild(i));
+});
+
+const form = document.createElement('form');
+
+form.classList.add('new-employee-form');
+document.body.append(form);
+
+const citiesOptions = [
+  'Tokyo',
+  'Singapore',
+  'London',
+  'New York',
+  'Edinburgh',
+  'San Francisco',
+];
+const formInputs = [
+  ['Name', 'text'],
+  ['Position', 'text'],
+  ['Age', 'number'],
+  ['Salary', 'number'],
+];
+
+function addInput(inputName, inputType) {
+  const label = document.createElement('label');
+  const input = document.createElement('input');
+
+  label.innerText = inputName + ': ';
+  input.type = inputType;
+  input.name = inputName.toLowerCase();
+  input.dataset.qa = inputName.toLowerCase();
+
+  label.append(input);
+
+  return label;
+};
+
+for (const [input, type] of formInputs) {
+  form.append(addInput(input, type));
+};
+
+form.querySelector(':nth-child(2)').after((() => {
+  const label = document.createElement('label');
+  const select = document.createElement('select');
+
+  label.innerText = 'Office: ';
+  select.name = 'office';
+  select.dataset.qa = 'office';
+
+  for (const city of citiesOptions) {
+    select.append(new Option(city));
+  }
+
+  label.append(select);
+
+  return label;
+})());
+
+const button = document.createElement('button');
+
+button.type = 'submit';
+button.innerText = 'Save to table';
+
+form.append(button);
+
+function saveDataToTable(data) {
+  const tr = document.createElement('tr');
+
+  tr.innerHTML = `
+    <td>${data.name}</td>
+    <td>${data.position}</td>
+    <td>${data.office}</td>
+    <td>${data.age}</td>
+    <td>$${Number(data.salary).toLocaleString('en-US')}</td>
+  `;
+
+  body.append(tr);
+};
+
+const pushNotification = (title, description, type) => {
+  const messageConteiner = document.createElement('div');
```

Comment:
```suggestion
  const messageContainer = document.createElement('div');
```

Reply:
typo

Reply:
I don’t understand how to make a check so that when the first letter is entered, the previous text of the cell is deleted, if I write in this place `initialValue` then only after removing the focus, the previous text is deleted, and the new one remains
![Desktop Screenshot 2022 11 20 - 17 02 38 94](https://user-images.githubusercontent.com/99988845/202909744-e5b5b475-6cd4-49a1-bf67-d7ea43b2b3ce.png)
![Desktop Screenshot 2022 11 20 - 17 06 08 46](https://user-images.githubusercontent.com/99988845/202909756-557b9014-f672-47e9-893d-b80ecb361f05.png)
![Desktop Screenshot 2022 11 20 - 17 06 28 46](https://user-images.githubusercontent.com/99988845/202909761-82d0e252-105c-4adb-89bd-40e3d0867740.png)

Reply:
I write my questions there because mentors in fe_chat don't know my all code and it will be very difficult explain to them my problems easily